debian/wview-mysql-create.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH WVIEW-MYSQL-CREATE 1 "November 19, 2009"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
wview-mysql-create \- script to create the MySQL Database, tables and user on local or remote server for wview export 
.SH SYNOPSIS
.B wview-mysql-create
.RI db_admin db_admin_passwd db_user db_passwd db_name
.br
.SH DESCRIPTION
This manual page describes the use of
.B wview-mysql-create
.
Create the MySQL Database, tables and user on local or remote server for wview export.
.BR    db_admin             - MySQL server admin with create and grant privileges.
.BR    db_admin_passwd      - MySQL server admin password.
.BR    db_user              - MySQL server user for export acivity (can be same as db_admin).
.BR    db_passwd            - MySQL server password.
.BR    db_name              - MySQL database name to create and grant privileges for.
.BR
.BR Note: REQUIRES: mysql CLI utility for database creation.
.BR Note: must be executed on the target MySQL server.
.SH OPTIONS
There are no options and db_admin, db_admin_passwd, db_user, db_passwd, db_name are all mandatory.
.TP
.SH SEE ALSO
.BR wview-mysql-export (1),
.BR /usr/share/doc/wview/wview-User-Manual.html
.br
.SH AUTHOR
wview-mysql-create was written by Mark Teel.
.PP
This manual page was written by Mark Teel <mteel2005@gmail.com>,
for the Debian project (and may be used by others).


ChangeLog
RELEASE VERSION HISTORY (newest to oldest)
------------------------------------------


5.21.5    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Remove debug message.




5.21.4    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Added stale pid file cleanup to the APT package start script.

2)  WMRUSB HID interface changes. Changed reader thread to blocking IO.
    Mitigated the console occasionally blasting out bogus data. Now zero
    packet loss. Fixed issue with pressure sanity check. Added sanity
    check for bogus rain accumulator values from the WMR console.




5.21.3    03-31-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Further WMRUSB interface improvements. Better packet framing.




5.21.2    03-30-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Updated debian APT postinst script to include copying the exfoliation skin
    for new installs.

2)  Enabled checksum processing for WMR stations.

3)  Changed LOOP windspeed items to floats for better precision.
    NOTE: If you use custom datafeed clients the LOOP data structure has changed.

4)  Added temp only packet support for WMRUSB stations. This adds support for
    add-on sensors such as the pool sensor.

5)  Fixed APT installs changing the wview.log file ownership incorrectly. Also
    restarts rsyslog automatically if a new install.




5.21.1    03-13-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Modified HILOW database storage so higher precision is used for cumulative
    data such as rain and ET.

2)  Fixed issue with yearly historical charts when generated at wview start
    (as opposed to at midnight). The current day was erroneously being included in
    the history data. This was most apparent with the UV and Solar Radiation
    charts.

3)  Fixed bug introduced in 5.21.0 release causing NULL wind direction to have
    360 added to it (to normalize calibration result). A value of -99640 was
    being stored/displayed.




5.21.0    03-05-2014
--------------------

radlib version required: 2.12.0 or newer

1)  WMRUSB improvements. Further reverse engineering of the undocumented protocol. Reset and
    heartbeat messages now working. Packet reception more stable. Initial packet acquisition
    faster resulting in faster startup.

2)  Modify CWOP packet header for new stricter compliance requirements of CWOP servers.

3)  Fix log message time_t bug for 64-bit platforms.

4)  Add timeout parameter for ssh rules. Added rows to config database.

5)  Updated data feed conversion logic to handle ARCHIVE_VALUE_NULL in loop data.

6)  Added flat line sensor value detection for outside temp, wind direction and wind speed.
    If there are no changes over a 4 hour period an email alert will be sent.

7)  Added Station Name, City and State to the wviewmgmt System Status page header.

8)  Added logic to SSH timer to fixup next timer interval if system time has skewed or
    if the ssh transfer took longer than 2 minutes.

9)  Fixed TE923 rain calculation to assume each tip is 1/36" instead of 1mm.

10) Fixed a bug for Vantage Pro stations which could cause the last archive interval of
    the previous day to be included in the current day HILOW data. This occurred if the
    midnight archive record was provided late by the console.

11) Added wet bulb temperature calculation and html tag --WetBulbTemp--. Added to all
    "readings.incx" templates.

12) Added email alert notification if no loop data has been received for a given archive
    interval. This applies to all stations except the Davis stations (alerts already
    exist for them).

13) Updated start scripts to detect and delete stale pid files.

14) Modified WH1080 data parsing to assume a humidity of 9% if the value from the
    station is undefined (0xff).




5.20.2    03-21-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added back "dailyrainin" for http reporting. They do no computation based on hourly rain values.

2)  Fixed a typo in the wviewmgmt station.php file. 




5.20.1    03-20-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Modified http rain submission (Wunderground and Weatherforyou) so only the
    "rainin" parameter is sent so daily rain will be correct. 




5.20.0    03-17-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added "wait_for_time_set" function to most wview start script examples so 
    wview daemons are not started until the system time is set. This applies to
    platforms without a real time clock which must reset their time during 
    startup.

2)  Updated the User Manual section on skin selection to better describe enabling
    extended data generation (UV, ET, radiation).

3)  Updated the debian post install script to try harder to find the default http
    document root location for creation of symbolic links to wviewmgmt and the
    weather home page.

4)  Added new configuration option to optionally disable using the Vantage Pro 
    console archive records and instead autogenerate them from the LOOP data.
    This is how all other stations work. It also supports non-standard console
    interfaces (those not using the datalogger on the console).

5)  Updated the Weatherforyou URL for data submissions.

6)  Fixed bug with DataFeed clients that caused an infinite loop if you
    restarted wview while there was a datafeed client connection.

7)  Fixed bug with Alarm Type configuration introduced in 5.19.0.

8)  Changed TWI station to compute rain fall rate as it is reported to not
    be provided by the station.

9)  Added socket retry logic to the station ethernet interface (used by the
    Vantage Pro IP datalogger and any serial-to-ethernet device).


dbexport/wview-mysql-create.sh
################################################################################
#
# File:           wview-mysql-create.sh
#
# Description:    Provide a script to create the wview database for MySQL.
#
# Usage:          wview-mysql-create <db_admin> <db_admin_passwd> <db_user> <db_passwd> <db_name>
#                 Arguments:
#                   db_admin       MySQL server admin with create and grant
#                                  privileges.
#                   db_admin_passwd MySQL server admin password.
#                   db_user        MySQL server user with create and grant
#                                  privileges.
#                   db_passwd      MySQL server password.
#                   db_name        MySQL database name to create and grant on.
#
# History:
# Engineer	  Date	    Ver   Comments
# MS Teel	  03/28/09   1    Original
#
################################################################################

show_usage()
{
    echo "USAGE:"
    echo "wview-mysql-create <db_admin> <db_admin_passwd> <db_user> <db_passwd> <db_name>"
    echo ""
    echo "    db_admin             - MySQL server admin with create and grant privileges."
    echo ""
    echo "    db_admin_passwd      - MySQL server admin password."
    echo ""
    echo "    db_user              - MySQL server user for export acivity (can be same as db_admin)."
    echo ""
    echo "    db_passwd            - MySQL server password."
    echo ""
    echo "    db_name              - MySQL database name to create and grant privileges for."
    echo ""
    echo "REQUIRES: mysql CLI utility for database creation."
    echo "MUST: be executed on the MySQL server."
    echo ""
}

################################################################################
#################################  M A C R O S  ################################
################################################################################

## DO NOT CHANGE ANY DEFS IN THIS BLOCK!

export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

## The defaults match the SQLite table names: 
SQL_ARCHIVE_TABLE=archive
SQL_HILOW_TABLE_INTEMP=inTemp
SQL_HILOW_TABLE_OUTTEMP=outTemp
SQL_HILOW_TABLE_INHUMIDITY=inHumidity
SQL_HILOW_TABLE_OUTHUMIDITY=outHumidity
SQL_HILOW_TABLE_BP=baromPressure
SQL_HILOW_TABLE_WINDSPEED=windSpeed
SQL_HILOW_TABLE_WINDGUST=windGust
SQL_HILOW_TABLE_DEWPOINT=dewPoint
SQL_HILOW_TABLE_RAIN=rain
SQL_HILOW_TABLE_RAINRATE=rainRate
SQL_HILOW_TABLE_WINDCHILL=windChill
SQL_HILOW_TABLE_HEATINDEX=heatIndex
SQL_HILOW_TABLE_ET=ET
SQL_HILOW_TABLE_UV=UV
SQL_HILOW_TABLE_SOLARRAD=solarRadiation
SQL_HILOW_TABLE_HAIL=hail
SQL_HILOW_TABLE_HAILRATE=hailRate
SQL_HILOW_TABLE_WINDDIR=windDir
SQL_NOAA_TABLE=noaaHistory

## Test command line arguments:
if [ "$1" = "help" ]; then
    show_usage
    exit 0
fi

## Check arguments:
if [ x"$1" = x ]; then
    echo "You must specify the MySQL admin username to create export database - use 'wview-mysql-create help' for details"
    show_usage
    exit 1
elif [ x"$2" = x ]; then
    echo "You must specify the MySQL admin password to create export database - use 'wview-mysql-create help' for details"
    show_usage
    exit 1
elif [ x"$3" = x ]; then
    echo "You must specify the MySQL username to create export database - use 'wview-mysql-create help' for details"
    show_usage
    exit 1
elif [ x"$4" = x ]; then
    echo "You must specify the MySQL password to create export database - use 'wview-mysql-create help' for details"
    show_usage
    exit 1
elif [ x"$5" = x ]; then
    echo "You must specify the MySQL database name to create export database - use 'wview-mysql-create help' for details"
    show_usage
    exit 1
else
    MYSQL_ADMIN_USERNAME=$1
    MYSQL_ADMIN_PASSWORD=$2
    MYSQL_USERNAME=$3
    MYSQL_PASSWORD=$4
    MYSQL_DBNAME=$5
fi

################################################################################
#######################  D E F I N E  F U N C T I O N S  #######################
################################################################################
syslog()
{
    logger -t WV_EXPORT $1 $2
}

syslog_error()
{
    logger -t WV_EXPORT_ERROR $1 $2
}

mysql_create()
{
    echo "create database $MYSQL_DBNAME;" > /tmp/commands.sql
    echo "grant all privileges on $MYSQL_DBNAME.* to $MYSQL_USERNAME@\"localhost\" identified by '$MYSQL_PASSWORD';" >> /tmp/commands.sql
    echo "grant all privileges on $MYSQL_DBNAME.* to $MYSQL_USERNAME@\"%\" identified by '$MYSQL_PASSWORD';" >> /tmp/commands.sql
    echo "use $MYSQL_DBNAME;" >> /tmp/commands.sql

    echo "create table $SQL_ARCHIVE_TABLE (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	usUnits INTEGER NOT NULL," >> /tmp/commands.sql
    echo "	arcInt INTEGER NOT NULL," >> /tmp/commands.sql
    echo "	barometer REAL," >> /tmp/commands.sql
    echo "	pressure REAL," >> /tmp/commands.sql
    echo "	altimeter REAL," >> /tmp/commands.sql
    echo "	inTemp REAL," >> /tmp/commands.sql
    echo "	outTemp REAL," >> /tmp/commands.sql
    echo "	inHumidity REAL," >> /tmp/commands.sql
    echo "	outHumidity REAL," >> /tmp/commands.sql
    echo "	windSpeed REAL," >> /tmp/commands.sql
    echo "	windDir REAL," >> /tmp/commands.sql
    echo "	windGust REAL," >> /tmp/commands.sql
    echo "	windGustDir REAL," >> /tmp/commands.sql
    echo "	rainRate REAL," >> /tmp/commands.sql
    echo "	rain REAL," >> /tmp/commands.sql
    echo "	dewpoint REAL," >> /tmp/commands.sql
    echo "	windchill REAL," >> /tmp/commands.sql
    echo "	heatindex REAL," >> /tmp/commands.sql
    echo "	ET REAL," >> /tmp/commands.sql
    echo "	radiation REAL," >> /tmp/commands.sql
    echo "	UV REAL," >> /tmp/commands.sql
    echo "	extraTemp1 REAL," >> /tmp/commands.sql
    echo "	extraTemp2 REAL," >> /tmp/commands.sql
    echo "	extraTemp3 REAL," >> /tmp/commands.sql
    echo "	soilTemp1 REAL," >> /tmp/commands.sql
    echo "	soilTemp2 REAL," >> /tmp/commands.sql
    echo "	soilTemp3 REAL," >> /tmp/commands.sql
    echo "	soilTemp4 REAL," >> /tmp/commands.sql
    echo "	leafTemp1 REAL," >> /tmp/commands.sql
    echo "	leafTemp2 REAL," >> /tmp/commands.sql
    echo "	extraHumid1 REAL," >> /tmp/commands.sql
    echo "	extraHumid2 REAL," >> /tmp/commands.sql
    echo "	soilMoist1 REAL," >> /tmp/commands.sql
    echo "	soilMoist2 REAL," >> /tmp/commands.sql
    echo "	soilMoist3 REAL," >> /tmp/commands.sql
    echo "	soilMoist4 REAL," >> /tmp/commands.sql
    echo "	leafWet1 REAL," >> /tmp/commands.sql
    echo "	leafWet2 REAL," >> /tmp/commands.sql
    echo "	rxCheckPercent REAL," >> /tmp/commands.sql
    echo "	txBatteryStatus REAL," >> /tmp/commands.sql
    echo "	consBatteryVoltage REAL," >> /tmp/commands.sql
    echo "	hail REAL," >> /tmp/commands.sql
    echo "	hailRate REAL," >> /tmp/commands.sql
    echo "	heatingTemp REAL," >> /tmp/commands.sql
    echo "	heatingVoltage REAL," >> /tmp/commands.sql
    echo "	supplyVoltage REAL," >> /tmp/commands.sql
    echo "	referenceVoltage REAL," >> /tmp/commands.sql
    echo "	windBatteryStatus REAL," >> /tmp/commands.sql
    echo "	rainBatteryStatus REAL," >> /tmp/commands.sql
    echo "	outTempBatteryStatus REAL," >> /tmp/commands.sql
    echo "	inTempBatteryStatus REAL" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_INTEMP (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_OUTTEMP (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_INHUMIDITY (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_OUTHUMIDITY (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_BP (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_WINDSPEED (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_WINDGUST (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_DEWPOINT (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_RAIN (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_RAINRATE (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_WINDCHILL (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_HEATINDEX (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_ET (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_UV (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_SOLARRAD (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_HAIL (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_HAILRATE (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	low REAL," >> /tmp/commands.sql
    echo "	timeLow INTEGER," >> /tmp/commands.sql
    echo "	high REAL," >> /tmp/commands.sql
    echo "	timeHigh INTEGER," >> /tmp/commands.sql
    echo "	whenHigh REAL," >> /tmp/commands.sql
    echo "	cumulative REAL," >> /tmp/commands.sql
    echo "	samples INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_HILOW_TABLE_WINDDIR (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	bin0 INTEGER," >> /tmp/commands.sql
    echo "	bin1 INTEGER," >> /tmp/commands.sql
    echo "	bin2 INTEGER," >> /tmp/commands.sql
    echo "	bin3 INTEGER," >> /tmp/commands.sql
    echo "	bin4 INTEGER," >> /tmp/commands.sql
    echo "	bin5 INTEGER," >> /tmp/commands.sql
    echo "	bin6 INTEGER," >> /tmp/commands.sql
    echo "	bin7 INTEGER," >> /tmp/commands.sql
    echo "	bin8 INTEGER," >> /tmp/commands.sql
    echo "	bin9 INTEGER," >> /tmp/commands.sql
    echo "	bin10 INTEGER," >> /tmp/commands.sql
    echo "	bin11 INTEGER," >> /tmp/commands.sql
    echo "	bin12 INTEGER," >> /tmp/commands.sql
    echo "	bin13 INTEGER," >> /tmp/commands.sql
    echo "	bin14 INTEGER," >> /tmp/commands.sql
    echo "	bin15 INTEGER," >> /tmp/commands.sql
    echo "	bin16 INTEGER," >> /tmp/commands.sql
    echo "	bin17 INTEGER," >> /tmp/commands.sql
    echo "	bin18 INTEGER," >> /tmp/commands.sql
    echo "	bin19 INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    echo "create table $SQL_NOAA_TABLE (" >> /tmp/commands.sql
    echo "	dateTime INTEGER NOT NULL PRIMARY KEY," >> /tmp/commands.sql
    echo "	meanTemp REAL," >> /tmp/commands.sql
    echo "	highTemp REAL," >> /tmp/commands.sql
    echo "	highTempTime INTEGER," >> /tmp/commands.sql
    echo "	lowTemp REAL," >> /tmp/commands.sql
    echo "	lowTempTime INTEGER," >> /tmp/commands.sql
    echo "	heatDegDays REAL," >> /tmp/commands.sql
    echo "	coolDegDays REAL," >> /tmp/commands.sql
    echo "	rain REAL," >> /tmp/commands.sql
    echo "	avgWind REAL," >> /tmp/commands.sql
    echo "	highWind REAL," >> /tmp/commands.sql
    echo "	highWindTime INTEGER," >> /tmp/commands.sql
    echo "	domWindDir INTEGER" >> /tmp/commands.sql
    echo ");" >> /tmp/commands.sql

    mysql --user=$MYSQL_ADMIN_USERNAME --password=$MYSQL_ADMIN_PASSWORD < /tmp/commands.sql > /tmp/mysql_create.log
    rm -rf /tmp/commands.sql
}

################################################################################
##################  S C R I P T  E X E C U T I O N  S T A R T  #################
################################################################################

mysql_create

## Done!
exit 0

